blocks/results-list-block/features/results-list/results/index.jsx

```diff
@@ -0,0 +1,326 @@
+      contentService,
+      contentConfigValues,
+    },
+  } = customFields;
+
+  const {
+    breakpoints,
+    fallbackImage,
+    locale,
+    primaryLogoAlt,
+    resizerURL,
+  } = getProperties(arcSite) || {};
+
+  const imageProps = {
+    smallWidth: 158,
+    smallHeight: 89,
+    mediumWidth: 274,
+    mediumHeight: 154,
+    largeWidth: 274,
+    largeHeight: 154,
+    primaryLogoAlt,
+    breakpoints,
+    resizerURL,
+  };
+
+  const configuredOffset = parseInt(contentConfigValues.offset, 10)
+    || parseInt(contentConfigValues.feedOffset, 10)
+    || parseInt(contentConfigValues.from, 10)
+    || 0;
+  const configuredSize = parseInt(contentConfigValues.size, 10)
+    || parseInt(contentConfigValues.feedSize, 10)
+    || 10;
+
+  const [queryOffset, setQueryOffset] = useState(configuredOffset);
+  const listItemRefs = useRef([]);
+
+  const targetFallbackImage = !(fallbackImage.includes('http'))
+    ? deployment(`${contextPath}/${fallbackImage}`)
+    : fallbackImage;
+
+  const placeholderResizedImageOptions = useContent({
+    source: !targetFallbackImage.includes('/resources/') ? 'resize-image-api' : null,
+    query: { raw_image_url: targetFallbackImage, respect_aspect_ratio: true },
+  });
+
+  const [resultList, alterResultList] = useReducer(reduceResultList, defaultResultList);
+  useEffect(() => {
+    const topItem = resultList.content_elements[queryOffset];
+    if (topItem?._id && listItemRefs.current[topItem?._id]) {
+      const focusLink = listItemRefs.current[topItem._id].querySelector('a:not([aria-hidden])');
+      if (focusLink) {
+        focusLink.focus();
+      }
+    }
+  }, [resultList, queryOffset]);
+
+  const serviceQueryPage = useCallback((requestedOffset = 0) => {
+    /*
+      This sets up a window view of the data starting from the initial index
+      to *twice* the size.  When clicking on the more button, we will render
+      the next size worth of items and load the next size from the server.
+    */
+    switch (contentService) {
+      case 'story-feed-query': {
+        const size = requestedOffset === configuredOffset
+          ? configuredSize * 2
+          : configuredSize;
+        const offset = requestedOffset === configuredOffset
+          ? configuredOffset
+          : requestedOffset + configuredSize;
+        return { offset, size };
+      }
+      case 'story-feed-author':
+      case 'story-feed-sections':
+      case 'story-feed-tag': {
+        const feedSize = requestedOffset === configuredOffset
+          ? configuredSize * 2
+          : configuredSize;
+        const feedOffset = requestedOffset === configuredOffset
+          ? configuredOffset
+          : requestedOffset + configuredSize;
+        return { feedOffset, feedSize };
+      }
+      case 'content-api-collections': {
+        const size = requestedOffset === configuredOffset
+          ? configuredSize * 2
+          : configuredSize;
+        const from = requestedOffset === configuredOffset
```

Comment:
Each `case` block works out the same size and offset. The only difference is the object retuned keys. We could reduce the duplication and just return in correct formats in the `case` blocks

Reply:
fixed in the latest